exchanges/bybit/bybit_websocket.go

```diff
@@ -792,3 +740,162 @@ const subTplText = `
+			authArg.associatedSubs = append(authArg.associatedSubs, s)
+		}
+
+		if len(arg.Arguments) >= 10 {
+			args = append(args, arg)
+			arg = SubscriptionArgument{
+				Operation: operation,
+				RequestID: strconv.FormatInt(conn.GenerateMessageID(false), 10),
+				Arguments: []string{},
+			}
+		}
+	}
+	if len(arg.Arguments) != 0 {
+		args = append(args, arg)
+	}
+	if len(authArg.Arguments) != 0 {
+		args = append(args, authArg)
+	}
+	return args, nil
+}
+
+// generateAuthSubscriptions generates default subscription for the dedicated auth websocket connection. These are
+// agnostic to the asset type and pair as all account level data will be routed through this connection.
+// TODO: Remove this function when template expansion is across all assets
+func (by *Bybit) generateAuthSubscriptions() (subscription.List, error) {
+	if !by.Websocket.CanUseAuthenticatedEndpoints() {
+		return nil, nil
+	}
+	var subscriptions subscription.List
+	for _, channel := range []string{chanPositions, chanExecution, chanOrder, chanWallet} {
+		subscriptions = append(subscriptions, &subscription.Subscription{
+			Channel: channel,
+			Pairs:   currency.Pairs{currency.EMPTYPAIR}, // This is a placeholder, the actual pair is not required for these channels
+			Asset:   asset.All,
+		})
+	}
+	return subscriptions, nil
+}
+
+// LinearSubscribe sends a subscription message to linear public channels.
+func (by *Bybit) authSubscribe(ctx context.Context, conn websocket.Connection, channelSubscriptions subscription.List) error {
+	return by.handleSubscriptionNonTemplate(ctx, conn, asset.Spot, "subscribe", channelSubscriptions)
+}
+
+// LinearUnsubscribe sends an unsubscription messages through linear public channels.
+func (by *Bybit) authUnsubscribe(ctx context.Context, conn websocket.Connection, channelSubscriptions subscription.List) error {
+	return by.handleSubscriptionNonTemplate(ctx, conn, asset.Spot, "unsubscribe", channelSubscriptions)
+}
+
+// getPairFromCategory returns the currency pair and asset type based on the category and symbol. Used with a dedicated
+// auth connection where multiple asset type changes are piped through a single connection.
+func (by *Bybit) getPairFromCategory(category, symbol string) (currency.Pair, asset.Item, error) {
+	assets := make([]asset.Item, 0, 2)
+	switch category {
+	case "spot":
+		assets = append(assets, asset.Spot)
+	case "inverse":
+		assets = append(assets, asset.CoinMarginedFutures)
+	case "linear":
+		assets = append(assets, asset.USDTMarginedFutures, asset.USDCMarginedFutures)
+	case "option":
+		assets = append(assets, asset.Options)
+	default:
+		return currency.EMPTYPAIR, 0, fmt.Errorf("category '%s' not supported for incoming symbol '%s'", category, symbol)
+	}
+	var err error
+	for _, a := range assets {
+		var cp currency.Pair
+		if cp, err = by.MatchSymbolWithAvailablePairs(symbol, a, hasPotentialDelimiter(a)); err == nil {
+			return cp, a, nil
+		}
+	}
+	return currency.EMPTYPAIR, 0, err
```

Comment:
```go
	for _, a := range assets {
		cp, err := by.MatchSymbolWithAvailablePairs(symbol, a, hasPotentialDelimiter(a))
		if err != nil {
			return currency.EMPTYPAIR, 0, err
		}
		return cp, a, nil
	}
	return currency.EMPTYPAIR, 0, nil
```
I think its an improvement to readability to not try funny business with err handling and such